paasta_tools/firewall_update.py

```diff
@@ -118,12 +119,21 @@ def smartstack_dependencies_of_running_firewalled_services(soa_dir=DEFAULT_SOA_D
 
         smartstack_dependencies = [d['smartstack'] for d in dependencies if d.get('smartstack')]
         for smartstack_dependency in smartstack_dependencies:
-            # TODO: filter down to only services that have no proxy_port
-            dependencies_to_services[smartstack_dependency].add((service, instance))
+            dependency_service, dependency_instance = smartstack_dependency.split('.', 1)
+
+            # filter down to only services that have no proxy_port
+            if service_proxy_port(dependency_service, soa_dir) is None:
+                dependencies_to_services[smartstack_dependency].add((service, instance))
```

Comment:
I kinda expected an "else" where the proxy is `not None` and and the corresponding firewall rule for yocalhost:proxy_port is in place?